### Análisis exploratorio de datos

In [4]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns



El análisis se lleva a cabo siguiendo los pasos que se detallan a continuación:

1) Definición del problema
2) Recopilación de datos
3) Análisis descriptivo
4) Limpieza de datos
5) Análisis de variables


#### 1) Definición del problema


Para este trabajo se utilizará un dataset proveniente del Sistema de Información Hídrica (SIH), el cual contiene información sobre los Derechos de Uso de Aguas otorgados por la Dirección Nacional de Aguas (DINAGUA) a personas físicas y jurídicas.

El SIH incluye datos asociados a obras de captación de aguas superficiales (tomas y embalses) y subterráneas (pozos). Los datos utilizados provienen una vez que el usuario solicita el derecho en el [Registro de usuarios para trámites de derechos de uso de aguas](https://www.gub.uy/tramites/registro-usuarios-tramites-derechos-uso-aguas).

Dado que el presente análisis se centra en los recursos hídricos subterráneos, se trabajará exclusivamente con los registros correspondientes a pozos.



Posteriormente, se realizará un proceso de exploración, limpieza y preparación de los datos para obtener un conjunto de datos adecuado para el entrenamiento de modelos de Machine Learning.


#### 2) Recopilación de datos


A continuación, se presenta una breve descripción de las variables incluidas en la base de datos, con el objetivo de facilitar la comprensión de la información disponible y su posterior análisis.


| Variable | Descripción |
|-----------|-------------|
| Nro. GEX | Número de expediente |
| Fecha Vencimiento | Vencimiento del derecho de uso de aguas públicas, máximo 10 años, pero depende del uso que se le va a dar al agua. |
| Estado | Estado de la solicitud ante Dinagua: en estudio, pendiente de resolución, registrada. |
| Regional | Oficina de Dinagua encargada del estudio de la solicitud. |
| Departamento | Ubicación de la perforación. |
| Fecha de Resolución | Fecha en la que se emite el derecho de uso de aguas públicas. |
| Caudal | m³/h de extracción de agua autorizados por Dinagua. |
| Volumen | Volumen anual de extracción de agua (Volumen = caudal × hs/día × día/mes × mes/año). |
| Uso | Clasificación general de cuál es el uso que se le va a dar al agua que se extrae. |
| Destino | Subcategoría de uso. Finalidad específica del agua que se extrae. |
| Latitud | Ubicación de la perforación. |
| Longitud | Ubicación de la perforación. |
| Mes Inicio | En qué mes del año se comienza con la extracción de agua. |
| Mes Final | En qué mes del año se finaliza con la extracción de agua. |
| Horas x Año | Cantidad de horas anuales de extracción de agua. |
| Depto.-Padrón | Ubicación de la perforación. |
| Tipo de Obra | En este caso el tipo de obra es perforación. |
| Id Solicitud | Identificador único. |
| Codigo Cuencas de nivel 5 | Cuencas hidrográficas. |
| Sección y localidad catastral | Sección y localidad catastral a la que pertenece el padrón. |
| Cuencas de nivel 3 | Cuencas hidrográficas. |
| Curso a Utilizar | Dato no útil. Relacionado con obras del tipo superficial. |
| Tipo de acuífero | Tipo de acuífero desde donde se obtiene el agua subterránea a extraer. Puede ser fisurado, sedimentario o ambos. |
| Profundidad | Profundidad en metros de la perforación. |
| Diámetro | Diámetro en milímetros de la perforación. |
| Caudal Ensayado | Caudal obtenido en el momento del ensayo de bombeo, medido en m³. |
| Nivel Estático | Profundidad a la que está el agua cuando el pozo no es utilizado por un período mayor a 8 horas, en relación al nivel de superficie. |
| Nivel Dinámico | Profundidad a la que está el agua cuando el pozo está en uso, en relación al nivel de superficie. |
| Caudal Máximo | Dato no útil. |
| Codigo Cuencas de nivel 2 | Cuencas hidrográficas. |
| Producción | Dato no útil. |

#### 2) Importación de dataset y creación del dataframe

In [5]:
df = pd.read_csv("../data/raw/Base_de_datos_Python_GRUPOA.csv", sep=";")

In [6]:
df

,Tipo de Solicitud,Nro. GEX,Fecha Vencimiento,Estado,Regional,Departamento,Tecnicos,Fecha de Solicitud,Fecha de Resolución,Fecha de Inscripción,...,profundidad,Diámetro,Caudal Ensayado,Nivel Estático,Nivel Dinámico,Caudal Máximo,Codigo Cuencas de nivel 2,Accion Solicitud,Tipo Resolución,Producción
0,Proyecto de Aprovechamiento,SGRH-00040200289-635.001,27/Mar/2011,Registrada,SALTO,SALTO,"Rodríguez, Jorge",18/set/2000,27/Mar/2001,NaN,...,55,"0,15","19,7",20,"37,5",NaN,14.0,Nueva,Autorización,7
1,Proyecto de Aprovechamiento,SGRH-00040200324-595.001,19/Oct/2010,Registrada,FRAY BENTOS,PAYSANDÚ,"Rodríguez, Jorge",10/Mar/2000,19/Oct/2000,25/Oct/2000,...,53,"0,15",9,25,29,NaN,18.0,Nueva,Autorización,4
2,Proyecto de Aprovechamiento,SGRH-00040200325-596.001,20/Oct/2010,Registrada,FRAY BENTOS,PAYSANDÚ,"Rodríguez, Jorge",10/Mar/2000,20/Oct/2000,27/Oct/2000,...,31,"0,15",25,2,5,NaN,18.0,Nueva,Autorización,4
3,Proyecto de Aprovechamiento,SGRH-00040200402-598.001,19/Oct/2010,Registrada,SAN JOSÉ,SAN JOSÉ,"Rodríguez, Jorge",11/Abr/2000,19/Oct/2000,25/Oct/2000,...,55,"0,15",20,28,"32,5",NaN,24.0,Nueva,Autorización,"2,5"
4,Proyecto de Aprovechamiento,SGRH-00040200414-609.001,10/Oct/2010,Registrada,SAN JOSÉ,CANELONES,"Rodríguez, Jorge",14/Abr/2000,10/Oct/2000,18/Oct/2000,...,64,"0,15","6,8","3,9","22,3",NaN,63.0,Nueva,Autorización,6
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
5695,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5696,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5697,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5698,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [7]:
df.shape

(5700, 38)

In [8]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 5700 entries, 0 to 5699
Data columns (total 38 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   Tipo de Solicitud              5159 non-null   str    
 1   Nro. GEX                       5157 non-null   str    
 2   Fecha Vencimiento              4700 non-null   str    
 3   Estado                         5159 non-null   str    
 4   Regional                       5157 non-null   str    
 5   Departamento                   5157 non-null   str    
 6   Tecnicos                       5142 non-null   str    
 7   Fecha de Solicitud             5158 non-null   str    
 8   Fecha de Resolución            4699 non-null   str    
 9   Fecha de Inscripción           4653 non-null   str    
 10  Caudal                         5113 non-null   str    
 11  Volumen                        5127 non-null   str    
 12  Uso                            5159 non-null   str    
 13 

In [9]:
df.describe().T

,count,mean,std,min,25%,50%,75%,max
Mes Inicio,5123.0,6.718915,5.342711,1.0,1.0,10.0,12.00,12.0
Mes Final,5123.0,9.740582,3.386649,0.0,10.0,11.0,12.00,12.0
Horas x Año,5121.0,3128.130443,1866.384298,1.0,1600.0,2880.0,4320.00,9000.0
Id Solicitud,5159.0,249482.421012,6255.075576,237902.0,244274.5,249362.0,254852.50,261225.0
Codigo Cuencas de nivel 5,5152.0,37711.344332,18888.650444,10105.0,22600.0,27400.0,58171.25,68710.0
Sección y localidad catastral,3387.0,5.623561,13.541791,0.0,2.0,4.0,7.00,746.0
Caudal Máximo,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Codigo Cuencas de nivel 2,5152.0,37.194682,18.909617,10.0,22.0,27.0,58.00,68.0



#### 3) Limpieza de datos

##### _- Eliminación de duplicados_


In [10]:
df.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
5695     True
5696     True
5697     True
5698     True
5699     True
Length: 5700, dtype: bool

In [11]:
df.duplicated().sum()

np.int64(540)

Se identifican 540 filas duplicadas dentro del conjunto de datos que representan aproximadamente el 9,47% del total de observaciones. A continuación se inspeccionan filas duplicadas para confirmarlas y entenderlas previo a eliminarlas.

In [12]:
df[df.duplicated()].head()

,Tipo de Solicitud,Nro. GEX,Fecha Vencimiento,Estado,Regional,Departamento,Tecnicos,Fecha de Solicitud,Fecha de Resolución,Fecha de Inscripción,...,profundidad,Diámetro,Caudal Ensayado,Nivel Estático,Nivel Dinámico,Caudal Máximo,Codigo Cuencas de nivel 2,Accion Solicitud,Tipo Resolución,Producción
5160,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5161,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5162,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5163,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5164,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [13]:
df.tail(10)

,Tipo de Solicitud,Nro. GEX,Fecha Vencimiento,Estado,Regional,Departamento,Tecnicos,Fecha de Solicitud,Fecha de Resolución,Fecha de Inscripción,...,profundidad,Diámetro,Caudal Ensayado,Nivel Estático,Nivel Dinámico,Caudal Máximo,Codigo Cuencas de nivel 2,Accion Solicitud,Tipo Resolución,Producción
5690,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5691,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5692,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5693,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5694,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5695,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5696,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5697,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5698,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5699,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


Parecería que las últimas filas del dataset están todas vacías y por lo tanto duplicadas entre si. A continuación se cuentan las filas completamente vacías para confirmar que las filas duplicadas están todas vacías.

In [14]:
df.isnull().all(axis=1).sum()

np.int64(541)


Se verifica la existencia de 541 filas completamente vacías sin información en ninguna de sus variables y, que además, generan 540 registros duplicados dentro del dataset. Por este motivo, se procede a su eliminación.

In [15]:
df = df.dropna(how="all")

In [16]:
print("Filas vacías:", df.isnull().all(axis=1).sum())
print("Duplicados:", df.duplicated().sum())
print("Dimensión:", df.shape)

Filas vacías: 0
Duplicados: 0
Dimensión: (5159, 38)


In [37]:
df

,Tipo de Solicitud,Nro. GEX,Fecha Vencimiento,Estado,Regional,Departamento,Tecnicos,Fecha de Solicitud,Fecha de Resolución,Fecha de Inscripción,...,Tipo de acuifero,profundidad,Diámetro,Caudal Ensayado,Nivel Estático,Nivel Dinámico,Codigo Cuencas de nivel 2,Accion Solicitud,Tipo Resolución,Producción
0,Proyecto de Aprovechamiento,SGRH-00040200289-635.001,2011-03-27,Registrada,SALTO,SALTO,"Rodríguez, Jorge",2000-09-18,2001-03-27,NaT,...,NaN,55.0,0.15,19.7,20.0,37.50,14.0,Nueva,Autorización,7
1,Proyecto de Aprovechamiento,SGRH-00040200324-595.001,2010-10-19,Registrada,FRAY BENTOS,PAYSANDÚ,"Rodríguez, Jorge",2000-03-10,2000-10-19,2000-10-25,...,NaN,53.0,0.15,9.0,25.0,29.00,18.0,Nueva,Autorización,4
2,Proyecto de Aprovechamiento,SGRH-00040200325-596.001,2010-10-20,Registrada,FRAY BENTOS,PAYSANDÚ,"Rodríguez, Jorge",2000-03-10,2000-10-20,2000-10-27,...,NaN,31.0,0.15,25.0,2.0,5.00,18.0,Nueva,Autorización,4
3,Proyecto de Aprovechamiento,SGRH-00040200402-598.001,2010-10-19,Registrada,SAN JOSÉ,SAN JOSÉ,"Rodríguez, Jorge",2000-04-11,2000-10-19,2000-10-25,...,NaN,55.0,0.15,20.0,28.0,32.50,24.0,Nueva,Autorización,"2,5"
4,Proyecto de Aprovechamiento,SGRH-00040200414-609.001,2010-10-10,Registrada,SAN JOSÉ,CANELONES,"Rodríguez, Jorge",2000-04-14,2000-10-10,2000-10-18,...,NaN,64.0,0.15,6.8,3.9,22.30,63.0,Nueva,Autorización,6
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
5154,Proyecto de Aprovechamiento,2026/36001/003138,NaT,En Estudio,DURAZNO,FLORES,MARGARITA MARÍA SIENRA VALLE,2026-03-23,NaT,NaT,...,FISURADO,35.0,150.00,8.0,3.2,4.65,57.0,Nueva,NaN,0
5155,Proyecto de Aprovechamiento,2026/36001/003236,NaT,En Estudio,FRAY BENTOS,PAYSANDÚ,ALEJANDRO LEONEL SCHIPILOV ZACCHINO,2026-03-24,NaT,NaT,...,AMBOS,48.0,160.00,7.0,5.0,17.00,18.0,Nueva,NaN,0
5156,Proyecto de Aprovechamiento,2026/36001/003447,NaT,En Estudio,FRAY BENTOS,COLONIA,Gustavo Norberto Galmarini Silva,2026-04-06,NaT,NaT,...,AMBOS,33.0,150.00,2.5,6.0,25.00,23.0,Nueva,NaN,0
5157,Proyecto de Aprovechamiento,2026/36001/003411,NaT,En Estudio,SALTO,SALTO,ALEJANDRO LEONEL SCHIPILOV ZACCHINO,2026-03-27,NaT,NaT,...,AMBOS,61.0,160.00,7.0,6.5,18.00,13.0,Renov/Modif,NaN,0


Tras la limpieza, el dataset queda conformado por 5.159 registros y 38 columnas, sin filas completamente vacías ni registros duplicados.

**_- Eliminación de información irrelevante_**

Como parte del proceso de exploración inicial, a continuación, se analiza la cantidad y el porcentaje de valores faltantes en cada columna del dataset. Esta evaluación permite identificar variables con información incompleta y determinar su potencial impacto en análisis posteriores.

In [18]:
faltantes = pd.DataFrame({
        'Cantidad': df.isnull().sum(),
    'Porcentaje': round(df.isnull().sum() / len(df) * 100, 2)})

faltantes.sort_values('Porcentaje', ascending=False)


,Cantidad,Porcentaje
Caudal Máximo,5159,100.00
Producción,3357,65.07
Curso a Utilizar,2584,50.09
Tipo de acuifero,2031,39.37
Sección y localidad catastral,1772,34.35
Fecha de Inscripción,506,9.81
Nivel Dinámico,504,9.77
Tipo Resolución,466,9.03
Fecha de Resolución,460,8.92
Fecha Vencimiento,459,8.90


La columa "Caudal Máximo" presenta un 100% de valores faltantes, por lo que no aporta información al análisis. Por lo tanto, se elimina dicha columna del conjunto de datos.

In [19]:
df = df.drop(columns=["Caudal Máximo"])

In [20]:
df.shape

(5159, 37)

Tras la eliminación de la variable "Caudal Máximo", el conjunto de datos queda conformado por 5.159 registros y 37 columnas

Por otro lado, la variable "Producción" es catalogada como un dato sin utilidad para el análisis desde la definición de las variables. Por este motivo, se decide eliminarla del conjunto de datos antes de continuar con el análisis exploratorio, evitando incorporar información irrelevante que pudiera afectar la interpretación de los resultados.

In [38]:
df = df.drop(columns=["Producción"])

In [39]:
df.shape

(5159, 36)

Tras la eliminación de la variable "Caudal Máximo", el conjunto de datos queda conformado por 5.159 registros y 36 columnas

_- Adecuación del formato de datos_

Durante el análisis inicial de la estructura del dataset, mediante la función df.info(), se identifica que varias variables tienen tipo de dato string (str). Sin embargo, de acuerdo a la definición de las variables y la inspección de los datos, se observa que algunas de ellas corresponden a fechas y otras representan magnitudes numéricas.

Esta situación puede limitar la realización de análisis estadísticos, cálculos y visualizaciones, ya que los tipos de datos no reflejan la naturaleza real de la información almacenada. Por este motivo, se procede a revisar el formato de dichas variables y a convertirlas al tipo de dato más adecuado, con el objetivo de garantizar la consistencia y calidad del conjunto de datos.

a) Adecuación de fechas

Para convertir correctamente las variables de fecha, se realiza una inspección de los formatos presentes en las columnas de fecha.

In [21]:
columnas_fecha = [
    'Fecha de Solicitud',
    'Fecha de Resolución',
    'Fecha de Inscripción',
    'Fecha Vencimiento'
]

In [22]:
for col in columnas_fecha:
    meses = (
        df[col]
        .dropna()
        .str.extract(r'/([A-Za-z]+)/')[0]
        .dropna()
        .unique()
    )

    print(f"\n{col}")
    print(meses)


Fecha de Solicitud
<ArrowStringArray>
['set', 'Mar', 'Abr', 'May', 'Jul', 'Nov', 'Dic', 'Ene', 'Jun', 'Ago', 'Oct',
 'Feb']
Length: 12, dtype: str

Fecha de Resolución
<ArrowStringArray>
['Mar', 'Oct', 'set', 'Dic', 'Jun', 'Abr', 'Jul', 'Nov', 'Ene', 'Ago', 'Feb',
 'May']
Length: 12, dtype: str

Fecha de Inscripción
<ArrowStringArray>
['Oct', 'set', 'Nov', 'Abr', 'Dic', 'Jun', 'Jul', 'Ene', 'Mar', 'Feb', 'Ago',
 'May']
Length: 12, dtype: str

Fecha Vencimiento
<ArrowStringArray>
['Mar', 'Oct', 'set', 'Jun', 'Dic', 'Abr', 'Jul', 'Nov', 'May', 'Ene', 'Feb',
 'Ago', 'Apr', 'Sep', 'Aug']
Length: 15, dtype: str


Se identifica que la mayoría de las fechas utilizan abreviaturas en español (Ene, Abr, Ago, Dic). Sin embargo, se detectaron algunos registros aislados con abreviaturas en inglés, tales como Apr, Aug y Sep.

Dado que la coexistencia de diferentes nomenclaturas puede generar inconsistencias durante el proceso de conversión, se estandarizan las abreviaturas de los meses antes de transformar las variables al formato fecha.

In [23]:
meses = {
    'Ene': 'Jan',
    'Feb': 'Feb',
    'Mar': 'Mar',
    'Abr': 'Apr',
    'May': 'May',
    'Jun': 'Jun',
    'Jul': 'Jul',
    'Ago': 'Aug',
    'set': 'Sep',
    'Oct': 'Oct',
    'Nov': 'Nov',
    'Dic': 'Dec'
}

columnas_fecha = [
    'Fecha de Solicitud',
    'Fecha de Resolución',
    'Fecha de Inscripción',
    'Fecha Vencimiento'
]

for col in columnas_fecha:
    for esp, eng in meses.items():
        df[col] = df[col].str.replace(esp, eng, regex=False)

In [24]:
for col in columnas_fecha:
    df[col] = pd.to_datetime(
        df[col],
        format='%d/%b/%Y',
        errors='coerce'
    )

In [25]:
df[columnas_fecha].dtypes

Fecha de Solicitud      datetime64[us]
Fecha de Resolución     datetime64[us]
Fecha de Inscripción    datetime64[us]
Fecha Vencimiento       datetime64[us]
dtype: object

In [26]:
df[columnas_fecha].head()

,Fecha de Solicitud,Fecha de Resolución,Fecha de Inscripción,Fecha Vencimiento
0,2000-09-18,2001-03-27,NaT,2011-03-27
1,2000-03-10,2000-10-19,2000-10-25,2010-10-19
2,2000-03-10,2000-10-20,2000-10-27,2010-10-20
3,2000-04-11,2000-10-19,2000-10-25,2010-10-19
4,2000-04-14,2000-10-10,2000-10-18,2010-10-10


In [27]:
df

,Tipo de Solicitud,Nro. GEX,Fecha Vencimiento,Estado,Regional,Departamento,Tecnicos,Fecha de Solicitud,Fecha de Resolución,Fecha de Inscripción,...,Tipo de acuifero,profundidad,Diámetro,Caudal Ensayado,Nivel Estático,Nivel Dinámico,Codigo Cuencas de nivel 2,Accion Solicitud,Tipo Resolución,Producción
0,Proyecto de Aprovechamiento,SGRH-00040200289-635.001,2011-03-27,Registrada,SALTO,SALTO,"Rodríguez, Jorge",2000-09-18,2001-03-27,NaT,...,NaN,55,"0,15","19,7",20,"37,5",14.0,Nueva,Autorización,7
1,Proyecto de Aprovechamiento,SGRH-00040200324-595.001,2010-10-19,Registrada,FRAY BENTOS,PAYSANDÚ,"Rodríguez, Jorge",2000-03-10,2000-10-19,2000-10-25,...,NaN,53,"0,15",9,25,29,18.0,Nueva,Autorización,4
2,Proyecto de Aprovechamiento,SGRH-00040200325-596.001,2010-10-20,Registrada,FRAY BENTOS,PAYSANDÚ,"Rodríguez, Jorge",2000-03-10,2000-10-20,2000-10-27,...,NaN,31,"0,15",25,2,5,18.0,Nueva,Autorización,4
3,Proyecto de Aprovechamiento,SGRH-00040200402-598.001,2010-10-19,Registrada,SAN JOSÉ,SAN JOSÉ,"Rodríguez, Jorge",2000-04-11,2000-10-19,2000-10-25,...,NaN,55,"0,15",20,28,"32,5",24.0,Nueva,Autorización,"2,5"
4,Proyecto de Aprovechamiento,SGRH-00040200414-609.001,2010-10-10,Registrada,SAN JOSÉ,CANELONES,"Rodríguez, Jorge",2000-04-14,2000-10-10,2000-10-18,...,NaN,64,"0,15","6,8","3,9","22,3",63.0,Nueva,Autorización,6
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
5154,Proyecto de Aprovechamiento,2026/36001/003138,NaT,En Estudio,DURAZNO,FLORES,MARGARITA MARÍA SIENRA VALLE,2026-03-23,NaT,NaT,...,FISURADO,35,150,8,"3,2","4,65",57.0,Nueva,NaN,0
5155,Proyecto de Aprovechamiento,2026/36001/003236,NaT,En Estudio,FRAY BENTOS,PAYSANDÚ,ALEJANDRO LEONEL SCHIPILOV ZACCHINO,2026-03-24,NaT,NaT,...,AMBOS,48,160,7,5,17,18.0,Nueva,NaN,0
5156,Proyecto de Aprovechamiento,2026/36001/003447,NaT,En Estudio,FRAY BENTOS,COLONIA,Gustavo Norberto Galmarini Silva,2026-04-06,NaT,NaT,...,AMBOS,33,150,"2,5",6,25,23.0,Nueva,NaN,0
5157,Proyecto de Aprovechamiento,2026/36001/003411,NaT,En Estudio,SALTO,SALTO,ALEJANDRO LEONEL SCHIPILOV ZACCHINO,2026-03-27,NaT,NaT,...,AMBOS,61,160,7,"6,5",18,13.0,Renov/Modif,NaN,0


Luego de estandarizadar las abreviaturas de los meses, se procedió a convertir las variables de fecha al tipo de dato datetime. Esta transformación permite que las fechas sean interpretadas correctamente en la realización de análisis temporales y la generación de visualizaciones cronológicas. Posteriormente, se verificó que las cuatro variables fueran almacenadas correctamente con formato fecha.

b) Adecuación de magnitudes numéricas

Una vez convertidas las variables temporales, se procede a revisar las variables que, de acuerdo con su definición, representan magnitudes numéricas pero que tienen formato texto. Se consideran para esta adecuación de formato las variables cuantitativas sobre las que resulta posible realizar operaciones matemáticas y análisis estadísticos.

Por lo tanto, se consideran para su transformación las variables Caudal, Volumen, Latitud, Longitud, Profundidad, Diámetro, Caudal Ensayado, Nivel Estático y Nivel Dinámico. Los códigos identificadores y códigos de cuenca se mantienen en su formato original, dado que representan categorías o identificadores y no magnitudes numéricas susceptibles de análisis estadístico.

Previo a la conversión, se inspecciona el formato de los registros con el objetivo de identificar posibles inconsistencias, como el uso de distintos separadores decimales o la presencia de caracteres no numéricos, que puedan afectar el proceso de transformación.

In [28]:
columnas_numericas = [
    'Caudal',
    'Volumen',
    'Latitud',
    'Longitud',
    'profundidad',
    'Diámetro',
    'Caudal Ensayado',
    'Nivel Estático',
    'Nivel Dinámico'
]

for col in columnas_numericas:
    print(f"\n--- {col} ---")
    print(df[col].dropna().unique()[:10])


--- Caudal ---
<ArrowStringArray>
['19,7', '9', '25', '20', '6,8', '3', '8', '18', '6', '12,8']
Length: 10, dtype: str

--- Volumen ---
<ArrowStringArray>
['35460', '16200', '16000',  '8000',  '7000',  '3500', '10800', '22500',
  '7500', '14000']
Length: 10, dtype: str

--- Latitud ---
<ArrowStringArray>
['-31,31716746', '-32,34177153', '-32,36978549', '-34,68004373',
  '-34,5045261', '-34,52060124',  '-34,7569185',  '-32,2973834',
 '-32,96808761', '-32,25813999']
Length: 10, dtype: str

--- Longitud ---
<ArrowStringArray>
['-57,95376019', '-58,08282352', '-57,87001899', '-56,73112873',
 '-56,30396653', '-56,22510334', '-56,30549516', '-57,80362401',
  '-58,0426651', '-58,02078585']
Length: 10, dtype: str

--- profundidad ---
<ArrowStringArray>
['55', '53', '31', '64', '50', '36', '58', '79', '134', '60']
Length: 10, dtype: str

--- Diámetro ---
<ArrowStringArray>
['0,15', '0,2', '0,14', '0,22', '0,11', '0,1', '0,24', '203', '0,4', '150']
Length: 10, dtype: str

--- Caudal Ensayado --

In [29]:
for col in columnas_numericas:
    invalidos = df[
        ~df[col].fillna("").str.match(r"^-?\d+[,.]?\d*$")
    ][col].dropna().unique()

    print(f"\n{col}")
    print(invalidos[:10])


Caudal
<ArrowStringArray>
[]
Length: 0, dtype: str

Volumen
<ArrowStringArray>
[]
Length: 0, dtype: str

Latitud
<ArrowStringArray>
['#¡NUM!']
Length: 1, dtype: str

Longitud
<ArrowStringArray>
[]
Length: 0, dtype: str

profundidad
<ArrowStringArray>
[]
Length: 0, dtype: str

Diámetro
<ArrowStringArray>
[]
Length: 0, dtype: str

Caudal Ensayado
<ArrowStringArray>
[]
Length: 0, dtype: str

Nivel Estático
<ArrowStringArray>
[]
Length: 0, dtype: str

Nivel Dinámico
<ArrowStringArray>
[]
Length: 0, dtype: str


In [30]:
(df["Latitud"] == "#¡NUM!").sum()

np.int64(1)

In [31]:
df[df["Latitud"] == "#¡NUM!"]

,Tipo de Solicitud,Nro. GEX,Fecha Vencimiento,Estado,Regional,Departamento,Tecnicos,Fecha de Solicitud,Fecha de Resolución,Fecha de Inscripción,...,Tipo de acuifero,profundidad,Diámetro,Caudal Ensayado,Nivel Estático,Nivel Dinámico,Codigo Cuencas de nivel 2,Accion Solicitud,Tipo Resolución,Producción
4772,Proyecto de Aprovechamiento,2025/36001/018383,NaT,Pendiente de Resolución,NaN,NaN,Viviana Leticia González Molina,2025-11-17,NaT,NaT,...,FISURADO,45,160,"2,9","-2,54","-8,09",60.0,Nueva,NaN,0


No se identifican valores incompatibles con una representación numérica, a excepción de un registro en la variable Latitud que contenía el valor #¡NUM!, correspondiente a un error de origen. Dado que dicho valor no representa una observación válida, se reemplaza por un valor nulo antes de realizar la transformación de las variables al formato numérico.

In [32]:
df["Latitud"] = df["Latitud"].replace("#¡NUM!", np.nan)

In [33]:
for col in columnas_numericas:
    df[col] = pd.to_numeric(
        df[col].str.replace(",", ".", regex=False),
        errors="coerce"
    )

In [34]:
df[columnas_numericas].dtypes

Caudal             float64
Volumen            float64
Latitud            float64
Longitud           float64
profundidad        float64
Diámetro           float64
Caudal Ensayado    float64
Nivel Estático     float64
Nivel Dinámico     float64
dtype: object

In [35]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 5159 entries, 0 to 5158
Data columns (total 37 columns):
 #   Column                         Non-Null Count  Dtype         
---  ------                         --------------  -----         
 0   Tipo de Solicitud              5159 non-null   str           
 1   Nro. GEX                       5157 non-null   str           
 2   Fecha Vencimiento              4685 non-null   datetime64[us]
 3   Estado                         5159 non-null   str           
 4   Regional                       5157 non-null   str           
 5   Departamento                   5157 non-null   str           
 6   Tecnicos                       5142 non-null   str           
 7   Fecha de Solicitud             5158 non-null   datetime64[us]
 8   Fecha de Resolución            4699 non-null   datetime64[us]
 9   Fecha de Inscripción           4653 non-null   datetime64[us]
 10  Caudal                         5113 non-null   float64       
 11  Volumen                     

In [41]:
df

,Tipo de Solicitud,Nro. GEX,Fecha Vencimiento,Estado,Regional,Departamento,Tecnicos,Fecha de Solicitud,Fecha de Resolución,Fecha de Inscripción,...,Curso a Utilizar,Tipo de acuifero,profundidad,Diámetro,Caudal Ensayado,Nivel Estático,Nivel Dinámico,Codigo Cuencas de nivel 2,Accion Solicitud,Tipo Resolución
0,Proyecto de Aprovechamiento,SGRH-00040200289-635.001,2011-03-27,Registrada,SALTO,SALTO,"Rodríguez, Jorge",2000-09-18,2001-03-27,NaT,...,148,NaN,55.0,0.15,19.7,20.0,37.50,14.0,Nueva,Autorización
1,Proyecto de Aprovechamiento,SGRH-00040200324-595.001,2010-10-19,Registrada,FRAY BENTOS,PAYSANDÚ,"Rodríguez, Jorge",2000-03-10,2000-10-19,2000-10-25,...,183,NaN,53.0,0.15,9.0,25.0,29.00,18.0,Nueva,Autorización
2,Proyecto de Aprovechamiento,SGRH-00040200325-596.001,2010-10-20,Registrada,FRAY BENTOS,PAYSANDÚ,"Rodríguez, Jorge",2000-03-10,2000-10-20,2000-10-27,...,183,NaN,31.0,0.15,25.0,2.0,5.00,18.0,Nueva,Autorización
3,Proyecto de Aprovechamiento,SGRH-00040200402-598.001,2010-10-19,Registrada,SAN JOSÉ,SAN JOSÉ,"Rodríguez, Jorge",2000-04-11,2000-10-19,2000-10-25,...,247,NaN,55.0,0.15,20.0,28.0,32.50,24.0,Nueva,Autorización
4,Proyecto de Aprovechamiento,SGRH-00040200414-609.001,2010-10-10,Registrada,SAN JOSÉ,CANELONES,"Rodríguez, Jorge",2000-04-14,2000-10-10,2000-10-18,...,635,NaN,64.0,0.15,6.8,3.9,22.30,63.0,Nueva,Autorización
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
5154,Proyecto de Aprovechamiento,2026/36001/003138,NaT,En Estudio,DURAZNO,FLORES,MARGARITA MARÍA SIENRA VALLE,2026-03-23,NaT,NaT,...,NaN,FISURADO,35.0,150.00,8.0,3.2,4.65,57.0,Nueva,NaN
5155,Proyecto de Aprovechamiento,2026/36001/003236,NaT,En Estudio,FRAY BENTOS,PAYSANDÚ,ALEJANDRO LEONEL SCHIPILOV ZACCHINO,2026-03-24,NaT,NaT,...,NaN,AMBOS,48.0,160.00,7.0,5.0,17.00,18.0,Nueva,NaN
5156,Proyecto de Aprovechamiento,2026/36001/003447,NaT,En Estudio,FRAY BENTOS,COLONIA,Gustavo Norberto Galmarini Silva,2026-04-06,NaT,NaT,...,NaN,AMBOS,33.0,150.00,2.5,6.0,25.00,23.0,Nueva,NaN
5157,Proyecto de Aprovechamiento,2026/36001/003411,NaT,En Estudio,SALTO,SALTO,ALEJANDRO LEONEL SCHIPILOV ZACCHINO,2026-03-27,NaT,NaT,...,139,AMBOS,61.0,160.00,7.0,6.5,18.00,13.0,Renov/Modif,NaN



#### 5) Análisis de variables


Una variable categórica es una variable que puede tomar uno de un número limitado de categorías o grupos. En este caso las variables categóricas son tipo de solicitud, estado, regional, departamento, técnicos, uso, destino, tipo de obra, tipo de acuífero, acción solicitud y tipo de resolución